### ISSUE TITLE
How did you get into programming?

### ISSUE THREAD

**Issue by kentcdodds** [TheLarkInn/ama#2]

**Comment by TheLarkInn** [comment]
After I graduate college with a Parish Music degree, I moved to Nebraska where my wife is from, and while looking for a viable career (protip: unless you are extremely talented, you cant do much with that degree), I decided to take a technical support role at MacPractice, Inc. This is a SMB/Startup that make native Mac medical management software. 

At heart, I am an extremely personable person, and quite the talker, so after about a year, breaking all support records, a couple awesome raises and becoming the most skill support representative for this software in house, I hit a brick wall. One day I realized that, constantly, I was having to work people around the same software defects in our product. I wanted to fix peoples problems, rather than work-folks around them.

This was the motivation initially. So I started learning programming by applying my work directly to my support role. First started by learning how to write AppleScript to use our computer softphone and automatically dial support numbers with just a keystroke. 

As I learned it more I started to speak with developers in-house (Obj-C) and one suggested I look into Ruby. I learned as much as I could from ruby, and really learned to love the non verbose, duck typed language. I realized quickly that this was something that excited me, and that I could pickup. 

I expressed interest in becoming an Obj-C engineer and was asked to start learning it. After another 1.5 years of side by side tech support (meeting all quotas) & QA'ing software, I was learning Obj-C. I then lobbied my boss and engineering VP if they would consider me promotable to this role. 

I was told yes, but was given a series of tests to accomplish before it would have to happen. After a few months of dragging their feet on the approval process, I had completed the tasks (implementing a live chat system (combination of OS ViewControllers, WebViews and JavaScript)), and earned the promotion. 

Then I had the opportunity (and now title), to work in a few other places, drop in my lap.

tl;dr look at my linked in profile for the rest of my job history.

I've now reached my 3rd official year of programming and 2nd full year of ft JavaScript.
